pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -140,16 +140,23 @@ def synchronize(remote_pk, repository_pk, mirror, skip_types):
             path = f"{repodata}/"
             new_url = urljoin(remote.url, path)
             if repodata_exists(remote, new_url):
-                stage = RpmFirstStage(remote, deferred_download, new_url=new_url)
+                stage = RpmFirstStage(remote, repository, deferred_download, new_url=new_url)
                 dv = RpmDeclarativeVersion(first_stage=stage,
                                            repository=sub_repo)
                 dv.create()
 
-    first_stage = RpmFirstStage(remote, deferred_download, skip_types, treeinfo=treeinfo)
+    first_stage = RpmFirstStage(remote,
+                                repository,
+                                deferred_download,
+                                skip_types,
+                                treeinfo=treeinfo)
     dv = RpmDeclarativeVersion(first_stage=first_stage,
                                repository=repository,
                                mirror=mirror)
     dv.create()
+    repository.remote_id = remote_pk
+    repository.last_sync_repo_version = repository.latest_version().number
+    repository.remote_last_updated = remote.pulp_last_updated
```

Comment:
You need to call repository.save() in order to persist these changes in the database.

Reply:
You don't need to keep track of remote.pulp_last_updated. You already know what repository version was created by the last sync. That repository version should have a pulp_created date on it. In the optimization code you need to ensure that the remote.pulp_last_updated is not a more recent date than the repository_version.pulp_created date. What do you think about that?

Reply:
@dkliban , if we are not storing remote_last_updated, then the order of the checks becomes very important (there can be other changes than sync, like adding/removing content, so the date or a repo version will be different from the last sync, so checking the version number should happen first then). If you go that path, please leave a clarifying comment in the code so any later optimisations or refactors don't screw up the logic.

Reply:
@goosemania  I completely agree. 

1. Check if remote is the same as last time 
2. Check if the repository version is the same as what was recorded last time.
3. Check if remote has been updated since the last repository version was created.
4. Check if the repomd revision number is the same.

We should skip syncing only if all of those are True in that order.